File: src/engine_zig/frame.zig
const Rectangle = @import("rectangle.zig").Rectangle;
const CollisionBox = @import("collision_box.zig").CollisionBox;

/// Estrutura que representa um único quadro (asset) de um Sprite
pub const Frame = struct {
    cut_rect: Rectangle,
    collision_box_list: []CollisionBox, // Slice nativo do Zig (ponteiro + tamanho)

    pub fn init(cut_rect: Rectangle, collision_box_list: []CollisionBox) Frame {
        return .{
            .cut_rect = cut_rect,
            .collision_box_list = collision_box_list,
        };
    }
};

File: src/engine_ts/sprites/frame.ts
import { CollisionBox } from "./collision_box";

/** Estrutura que representa um único quadro de um Sprite */
export class Frame {
    public cutX!: number;
    public cutY!: number;
    public collisionBoxList: CollisionBox[]; // representa as caixas de colisão do quadro no Sprite

    constructor(cutX: number, cutY: number, collisionBoxList: CollisionBox[] | null) {
        this.cutX = cutX << 8;
        this.cutY = cutY << 8;
        this.collisionBoxList = collisionBoxList ? collisionBoxList : [];
    }
}